dolphin/drivers/dellemc_storage/vmax.py

```diff
@@ -0,0 +1,193 @@
+
+class VMAXStorageDriver(driver.StorageDriver):
+    """VMAXStorageDriver implement the DELL EMC Storage driver,
+    """
+    # Storage_id's that this driver manages
+    storage_ids = []
+
+    def __init__(self, storage_id):
+        super().__init__(storage_id)
+        # Initialize with defaults
+        self.storage_id = storage_id
+        self.univmax_version = '90'
+        self.server_ip = '127.0.0.1'
+        self.port = '8443'
+        self.verify = False  # Valid values are True/False/<path to certificatefile>
+        self.username = ''
+        self.password = ''
+        self.symmetrix_id = ''
+
+    @staticmethod
+    def get_storage_registry():
+        # register_info dict
+        extra_attributes = {
+            'symmetrix_id':self.symmetrix_id,
+            'univmax_version': self.univmax_version,
+            'server_ip': self.server_ip,
+            'port': self.port,
+            'verify': self.verify
+        }
+        register_info = {
+            'storage_id':self.storage_id,
+            'username':'',
+            'password':'',
+            'hostname':'',
+            'extra_attributes': extra_attributes
+        }
+        LOG.info('get_storage_registry(), typical input dictionary is', register_info)
+        return register_info
+
+    def register_storage(self, context, register_info):
+        # Check if the storage is already registered
+        if self.storage_id in VMAXStorageDriver.storage_ids:
+            LOG.info('Storage is already registered!')
+            return VMAXStorageDriver.storage_ids
+
+        # Verify input parameters for registration exists
+        try:
+            self.symmetrix_id = register_info['extra_attributes']['symmetrix_id']
+            self.univmax_version = register_info['extra_attributes']['univmax_version']
+            self.server_ip = register_info['extra_attributes']['server_ip']
+            self.port = register_info['extra_attributes']['port']
+            self.verify = register_info['extra_attributes']['verify']
+            self.username = register_info['username']
+            self.password = register_info['password']
+        except KeyError as err:
+            LOG.error('ERROR: Invalid input register_info[' + str(err) + ']')
+            return []
+
+        try:
+            # Initialise PyU4V connection to Unisphere
+            conn = PyU4V.U4VConn(
+                univmax_version=self.univmax_version,
+                server_ip=self.server_ip,
+                port=self.port,
+                verify=self.verify,
+                username=self.username,
+                password=self.password)
+
+            # Get the Unisphere version
+            version = conn.common.get_uni_version()
+
+            # Retrieve a list of arrays managed by your instance of Unisphere
+            array_list = conn.common.get_array_list()
+
+            # Close the session
+            conn.close_session()
+
+            if (version != self.univmax_version):
```

Comment:
version need not be checked against user input. But can be checked against a supported unisphere version list.